src/state/slices/opportunitiesSlice/resolvers/cosmosSdk/utils.ts

```diff
@@ -33,3 +41,44 @@ export const makeUniqueValidatorAccountIds = (
       ),
     ]),
   )
+
+export const makeAccountUserData = ({
+  cosmosAccount,
+  validatorIds,
+}: {
+  cosmosAccount: Account<CosmosSdkChainId>
+  validatorIds: ValidatorId[]
+}): OpportunitiesState['userStaking']['byId'] => {
+  const delegations = cosmosAccount.chainSpecific.delegations
+  // const undelegations = cosmosAccount.chainSpecific.delegations
+  const rewards = cosmosAccount.chainSpecific.rewards
+
+  const delegationsByValidator = groupBy(delegations, delegation => delegation.validator.address)
+  // TODO(gomes): handle me in a separate PR, we will need chain-specific userData similar to the savers-specific-metadata
+  // const undelegationsByValidator = groupBy(
+  // undelegations,
+  // undelegation => undelegation.validator.address,
+  // )
+  const rewardsByValidator = groupBy(rewards, reward => reward.validator.address)
+
+  return validatorIds.reduce((acc, validatorId) => {
+    const validatorAddress = fromAccountId(validatorId).account
+    const userStakingId = serializeUserStakingId(
+      toAccountId({ account: cosmosAccount.pubkey, chainId: cosmosAccount.chainId }),
+      validatorId,
+    )
+
+    const maybeValidatorDelegations = delegationsByValidator[validatorAddress][0]
+    const maybeValidatorRewards = rewardsByValidator[validatorAddress]?.[0]?.rewards
+    const maybeValidatorRewardsAggregated = maybeValidatorRewards
+      ? maybeValidatorRewards.reduce((a, b) => a.plus(b.amount), bn(0)).toFixed()
+      : '0'
+
+    acc[userStakingId] = {
+      stakedAmountCryptoBaseUnit: maybeValidatorDelegations?.amount ?? '0',
+      rewardsAmountsCryptoBaseUnit: [maybeValidatorRewardsAggregated ?? '0'],
```

Comment:
Don't insert empty userData, since `userStakingId` will most likely be used to detect an "active" opportunity

Reply:
803d2d3a0487392c91d0cd447b14c1d50b548910